spring-cloud-gcp-starters/spring-cloud-gcp-starter-storage/src/main/java/org/springframework/cloud/gcp/storage/autoconfig/StorageAutoConfiguration.java

```diff
@@ -31,40 +31,47 @@
 import org.springframework.cloud.gcp.core.GcpScope;
 import org.springframework.cloud.gcp.storage.GcpStorageProperties;
 import org.springframework.cloud.gcp.storage.GoogleStorageProtocolResolver;
+import org.springframework.cloud.gcp.storage.GoogleStorageProtocolResolverSettings;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Import;
 
 /**
- * An auto-configuration for Google {@link Storage} bean definition. Also it
- * {@link Import} a {@link GoogleStorageProtocolResolver} to register it
- * with the {@code DefaultResourceLoader}.
+ * An auto-configuration for Google {@link GoogleStorageProtocolResolverSettings} bean
+ * definition. Also it {@link Import} a {@link GoogleStorageProtocolResolver} to register
+ * it with the {@code DefaultResourceLoader}.
  *
  * @author Vinicius Carvalho
  * @author Artem Bilan
  *
  * @see GoogleStorageProtocolResolver
  */
 @Configuration
-@ConditionalOnClass(Storage.class)
+@ConditionalOnClass({ GoogleStorageProtocolResolverSettings.class, Storage.class })
 @EnableConfigurationProperties({GcpProperties.class, GcpStorageProperties.class})
 @Import(GoogleStorageProtocolResolver.class)
 public class StorageAutoConfiguration {
 
 	@Bean
 	@ConditionalOnMissingBean
-	public static Storage storage(
-			CredentialsProvider credentialsProvider,
+	public static Storage storage(CredentialsProvider credentialsProvider,
 			GcpStorageProperties gcpStorageProperties) throws IOException {
 		return StorageOptions.newBuilder()
 				.setCredentials(gcpStorageProperties.getCredentialsLocation() != null
-						? GoogleCredentials.fromStream(
-								gcpStorageProperties.getCredentialsLocation().getInputStream())
-						.createScoped(Collections.singletonList(
-								GcpScope.STORAGE_READ_WRITE.getUrl()))
+						? GoogleCredentials
+								.fromStream(gcpStorageProperties.getCredentialsLocation()
+										.getInputStream())
+								.createScoped(Collections.singletonList(
+										GcpScope.STORAGE_READ_WRITE.getUrl()))
 						: credentialsProvider.getCredentials())
-				.build()
-				.getService();
+				.build().getService();
 	}
 
+	@Bean
+	@ConditionalOnMissingBean
+	public static GoogleStorageProtocolResolverSettings googleStorageProtocolResolverContext(
```

Comment:
why do we need a new Settings object? can we not just re-use the gcp storage properties?

Reply:
The properties object is in the stater and the core storage module doesn't depend on the starter. I'm not sure if it *has* to be that way though.

Reply:
The Gcp storage properties is in the starter project, which depends on the cloud storage project, and to use the gcp storage properties in the cloud storage project would mean a circular dependency.

Reply:
hmm this is where we really ought to put properties and autoconfig into the same package as most of the other spring starters do.